os/binfmt/libelf/libelf_sections.c

```diff
@@ -65,13 +65,14 @@
 #include <tinyara/kmalloc.h>
 #include <tinyara/binfmt/elf.h>
 
-#include <queue.h>
 #include <string.h>
 #include <tinyara/binfmt/binfmt.h>
 #include <binary_manager/binary_manager.h>
 
 #include "libelf.h"
 
+static bin_addr_info_t g_bin_addr_list[CONFIG_NUM_APPS + 1];
```

Comment:
Could you let me know why one more additional array size is necessary? `+ 1`?

Reply:
It needs additional space for common binary.

Reply:
Then, how about below?
```
#ifdef config_common_bin
#define NUM_LOADING_BIN  (CONFIG_NUM_APPS + 1)
#else
#define NUM_LOADING_BIN  (CONFIG_NUM_APPS)
#endif

static .... g_bin_addr_list[NUM_LOADING_BIN];
```

Reply:
```
#define NUM_LOADING_BIN (CONFIG_NUM_APPS \
#ifdef config_common_bin
               + 1 \
#endif
              )
```

Reply:
The index 0 in binary table is fixed for common binary to handle it differently from other binary.
```
/* Index of 'Common Library' data in binary table. */
#define BM_CMNLIB_IDX              0

/* Binary table, the first data [0] is for Common Library. */
static binmgr_uinfo_t bin_table[USER_BIN_COUNT + 1];
```